File: 333-Programming-Languages/notes/prompt.py
import sys

def main():
	'''
	Prompts for a digit until 0 is entered.
	'''
	val = -1;
	while val != 0:
		try:
			val = int(raw_input("Enter a number 0..9: "))
			
			if not (0 <= val <= 9):
				raise Exception(
					"Error number {0} is out of range (0..9)".
					format(val))
			
			print "\taccepted: {0}".format(val);
				
		except ValueError:
			sys.stderr.write("Error: value is not a number\n")
		
		except Exception as e:
			sys.stderr.write("Error: {0}\n".format(e))
		
	print "Terminating"
	
if __name__ == "__main__":
	main()

File: 333-Programming-Languages/notes/Prompt.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.io.IOException;

/**
 * Prompts for a digit until 0 is entered.
 */
public class Prompt
{
	public static void main(String[] args)
	{
		BufferedReader input = 
			new BufferedReader(
			new InputStreamReader(System.in));
		
		int val = -1;
		while (val != 0)
		{
			try
			{
				System.out.print("Enter a number 0..9: ");
				String line = input.readLine();
				val = Integer.parseInt(line);
				
				if (val < 0 || val > 9)
				{
					throw new Exception(String.format(
						"number %d is out of range (0..9)", val));
				}
				
				System.out.println(String.format(
					"\taccepted: %d", val));
			}
			catch (IOException ioe)
			{
				System.err.println(
					"Input not valid: " + ioe.getMessage());
			}
			catch (NumberFormatException nfe)
			{
				System.err.println(
					"Value is not a number: " + nfe.getMessage());
			}
			catch (Exception e)
			{
				System.err.println(
					"Error: " + e.getMessage());
			}
		}
		
		System.out.println("Terminating");
	}
}